pl_bolts/models/detection/yolo/torch_networks.py

```diff
@@ -0,0 +1,1264 @@
+        self.downsample5 = downsample(w5, w5)
+        self.pan6 = csp(w5 + w6, w6)
+
+        self.out3 = out(w3)
+        self.out4 = out(w4)
+        self.out5 = out(w5)
+        self.out6 = out(w6)
+
+        self.detect3 = detect(range(0, anchors_per_cell))
+        self.detect4 = detect(range(anchors_per_cell, anchors_per_cell * 2))
+        self.detect5 = detect(range(anchors_per_cell * 2, anchors_per_cell * 3))
+        self.detect6 = detect(range(anchors_per_cell * 3, anchors_per_cell * 4))
+
+    def forward(self, x: Tensor, targets: Optional[TARGETS] = None) -> NETWORK_OUTPUT:
+        detections = []  # Outputs from detection layers
+        losses = []  # Losses from detection layers
+        hits = []  # Number of targets each detection layer was responsible for
+
+        image_size = get_image_size(x)
+
+        c3, c4, c5, x = self.backbone(x)[-4:]
+        c6 = self.spp(x)
+
+        x = torch.cat((self.upsample6(c6), self.pre5(c5)), dim=1)
+        p5 = self.fpn5(x)
+        x = torch.cat((self.upsample5(p5), self.pre4(c4)), dim=1)
+        p4 = self.fpn4(x)
+        x = torch.cat((self.upsample4(p4), self.pre3(c3)), dim=1)
+        n3 = self.fpn3(x)
+        x = torch.cat((self.downsample3(n3), p4), dim=1)
+        n4 = self.pan4(x)
+        x = torch.cat((self.downsample4(n4), p5), dim=1)
+        n5 = self.pan5(x)
+        x = torch.cat((self.downsample5(n5), c6), dim=1)
+        n6 = self.pan6(x)
+
+        y = self.detect3(self.out3(n3), image_size, targets)
+        detections.append(y)
+        if targets is not None:
+            losses.append(self.detect3.losses)
+            hits.append(self.detect3.hits)
+
+        y = self.detect4(self.out4(n4), image_size, targets)
+        detections.append(y)
+        if targets is not None:
+            losses.append(self.detect4.losses)
+            hits.append(self.detect4.hits)
+
+        y = self.detect5(self.out5(n5), image_size, targets)
+        detections.append(y)
+        if targets is not None:
+            losses.append(self.detect5.losses)
+            hits.append(self.detect5.hits)
+
+        y = self.detect6(self.out6(n6), image_size, targets)
+        detections.append(y)
+        if targets is not None:
+            losses.append(self.detect6.losses)
+            hits.append(self.detect6.hits)
+
+        return detections, losses, hits
+
+
+class YOLOV5Network(nn.Module):
+    """The YOLOv5 network architecture. Different variants (n/s/m/l/x) can be achieved by adjusting the ``depth``
+    and ``width`` parameters.
+
+    Args:
+        num_classes: Number of different classes that this model predicts.
+        backbone: A backbone network that returns the output from each stage.
+        width: The number of channels in the narrowest convolutional layer. The wider convolutional layers will use a
+            number of channels that is a multiple of this value.
+        depth: Repeat the bottleneck layers this many times. Can be used to make the network deeper.
```

Comment:
A depth of 3 would represent which variant? Not sure if it is well documented somewhere else, but we could try to show depth-to-variant settings

Reply:
> I'm not sure if we are still doing it but adding `catch_warnings` to all tests would be helpful.
> 
> An idea for bolts in general would have a README for each model folder. I think we could write the training process of the different YOLO networks there, maybe upload pretrained weights, etc.

I added `catch_warnings` to all tests and a README.

Reply:
I listed the settings for the different variants.